src/toplc.ml

```diff
@@ -656,17 +651,84 @@ let get_ancestors h c =
   ga c;
   U.hashtbl_fold_keys (fun c cs -> c :: cs) cs []
 
-let name_of_class c =
-  B.Utils.UTF8.to_string (B.Name.external_utf8_for_class c)
-
-let mk_full_method_name c mn =
-  name_of_class c ^ "." ^ mn
-
 let get_overrides h c m =
   let ancestors = get_ancestors h c in
   let qualify c =  mk_full_method_name c m.method_name in
   (List.map qualify ancestors, m.method_arity)
 
+let pp_parameter f p  = Format.fprintf f "%s" (B.Utils.UTF8.to_string(BD.utf8_of_parameter(p)))
+
+let if_ b xs = if b then List.concat xs else []
+
+let has = (<>) None
+        
+let instrument_call locals parameters call_id instrs =
+  let rev_params = List.rev parameters in
+  let max = locals + List.length parameters in
+  let instrumentation =
+    if log_in then printf "Call Instrumented with call ID: %a, Locals: %i, Max: %i, Parameter List: %a\n"
+                   (Format.pp_print_option Format.pp_print_int) call_id locals max (Format.pp_print_list pp_parameter) parameters; 
+  List.concat
+  [ 
+      List.concat (List.mapi (fun i param -> bc_store (max - i) param) rev_params )
+    ; List.concat (List.mapi (fun i param -> bc_load (locals + 1 + i) param) parameters)
```

Comment:
Are locals counted from 0? If so, shouldn't the indices be `(max-i-1)` and `(locals+i)`, respectively?